ui/dialog/preferences.py

```diff
@@ -169,81 +158,63 @@ def _project_path_dialog(self):
-            param : str
-                Parameter name
-
-            """
-            value = 'True' if checkbox_widget.isChecked() else 'False'
-            logger.info(value)
-            configurations.update_setting(TOML_PATH, section, param, value)
-
-        apply_checkbox(self.descCheck, 'Settings', 'ShowDescriptionPanel')
-        apply_checkbox(self.debugCheck, 'Settings', 'ShowDebugLog')
-        apply_checkbox(self.boxPrefix, 'Assets', 'UsePrefix')
-        apply_checkbox(self.boxSuffix, 'Assets', 'UseSuffix')
-        apply_checkbox(self.metadataCheck, 'Advanced', 'UseMetadata')
-
-        def apply_line_value(line_widget: QtWidgets.QLineEdit, section: str, param: str):
-            """Save line value in TOML after apply.
-
-            Parameters
-            ----------
-            line_widget : QtWidgets.QLineEdit
-                QLineEdit instance
-            section : str
-                Section name
-            param : str
-                Parameter name
-
-            """
-            value = line_widget.text()
-            logger.info(value)
-            configurations.update_setting(TOML_PATH, section, param, value)
-
-        apply_line_value(self.projectPathLine, 'Settings', 'ProjectPath')
-        apply_line_value(self.suffixCustomName, 'Assets', 'SuffixCustomName')
-
-        def apply_list_value(list_widget: QtWidgets.QListWidget, section: str, param: str):
-            """Save list value in TOML  after apply.
-
-            Parameters
-            ----------
-            list_widget : QtWidgets.QListWidget
-                QListWidget instance
-            section : str
-                Section name
-            param : str
-                Parameter name
-
-            """
-            items = []
-            for x in range(list_widget.count()):
-                value = list_widget.item(x).text()
-                items.append(value)
-            logger.info(items)
-            configurations.update_setting(TOML_PATH, section, param, str(items))
-
-        apply_list_value(self.categoryList, "Assets", "CategoryList")
-        apply_list_value(self.subfolderList, "Assets", "SubfolderList")
-
-        def apply_theme():
-            value = str(self.themeBtnGrp.checkedButton().text())
-            logger.info(value)
-            configurations.update_setting(TOML_PATH, 'UI', 'Theme', value)
-
-        apply_theme()
-
+    def get_checkbox_value(self, widget: QtWidgets.QCheckBox, setting: str):
+        value = widget.isChecked()
+        config = {setting: value}
+        return config
+
+    def get_line_value(self, widget: QtWidgets.QLineEdit, setting: str):
+        value = widget.text()
+        config = {setting: value}
+        return config
+
+    def get_list_value(self, widget: QtWidgets.QListWidget, setting: str):
+        items = []
+        for x in range(widget.count()):
+            value = widget.item(x).text()
+            items.append(value)
+        config = {setting: items}
+        return config
```

Comment:
Hmm maybe can rewrite this to check for widget instance to reduce DRY.

Something like
```
def get_widget_value(self, widget, setting):
    if isinstance(widget, QtWidgets.QCheckBox):
        value = widget.isChecked()
    if isinstance(widget, QtWidgets.QLineEdit):
        value = widget.text()
    blah
```